File: EngagementHub/TOS.EngagementHub.Models/Project.cs
using MongoDB.Bson;
using System;
using System.Collections.Generic;

namespace TOS.EngagementHub.Models
{
    public class Project : EngagementModel
    {
        public string Name { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public Guid LeadId { get; set; }
        public ICollection<Guid> RoleIds { get; set; } = new List<Guid>();
        public Guid ClientId { get; set; }
        public Location Location { get; set; }
        public ProjectStatus Status { get; set; }
    }
}

File: NiceReads/Backend/Server/tos-server/src/application/models/Project.ts
import { Uuid } from '../../common/uuid';
import { Document } from './Document';
import { Location } from './Location';
import { ProjectStatus } from './ProjectStatus';

export class Project extends Document {
    constructor(
        public name: string  = '',
        public startDate: Date | undefined = undefined,
        public endDate: Date | undefined = undefined,
        public leadId : Uuid | undefined = undefined,
        public roleIds: Uuid[] = [],
        public clientId: Uuid | undefined = undefined,
        public location: Location | undefined = undefined,
        public status: ProjectStatus | undefined = undefined,
        id: Uuid | undefined = undefined
    ) {
        super(id);
    }
}